AgentDiscoveries-Frontend/app/src/components/search-forms/search-location-reports.js

```diff
@@ -0,0 +1,76 @@
+import * as React from "react";
+import {
+    Form,
+    FormGroup,
+    FormControl,
+    Button,
+    ControlLabel
+} from "react-bootstrap";
+
+import * as SearchUtils from "./search-utilities"
+
+import * as CRUD from "../crud"
+import SearchResult from "./search-result"
+import { searchAPI } from "../crud";
+export default class LocationReportsSearch extends React.Component {
+
+    constructor() {
+        super();
+        this.state = {
+            "searchForm": {},
+            "results": []
+        }
+    }
+
+    render() {
+        return (
+            <div className="col-md-12">
+                <Form onSubmit={this.handleSearch.bind(this)}>
+                    <h3>API Location Report Search</h3>
+
+                    <FormGroup>
+                        <ControlLabel>Agent ID</ControlLabel>
+                        <FormControl type="text"
+                            inputRef={agentId => this.state.searchForm.agentId = agentId}
+                            placeholder="enter agent ID" />
+
+                        <ControlLabel>Location ID</ControlLabel>
+                        <FormControl type="text"
+                            inputRef={locationId => this.state.searchForm.locationId = locationId}
+                            placeholder="enter location ID" />
+
+                        <ControlLabel>From</ControlLabel>
+                        <FormControl type="datetime-local"
+                            inputRef={fromTime => this.state.searchForm.fromTime = fromTime}
+                            defaultValue={SearchUtils.getFormDate(SearchUtils.getDateDaysAgo(7))}
+                        />
+
+                        <ControlLabel>To</ControlLabel>
+                        <FormControl type="datetime-local"
+                            inputRef={toTime => this.state.searchForm.toTime = toTime}
+                            defaultValue={SearchUtils.getFormDate(new Date())} />
+
+                        <Button type="submit">Search</Button>
+                    </FormGroup>
+                </Form>
+                
+                <SearchResult results={this.state.results} />
+            </div>
+        );
+    }
+
+    handleSearch(e) {
+        e.preventDefault();
+
+        const searchParams = Object.keys(this.state.searchForm).map((key) => {
+            return this.state.searchForm[key].value == "" ? "" : encodeURIComponent(key) + '=' + encodeURIComponent(SearchUtils.getTransformedData(key, this.state.searchForm[key].value));

```

Comment:
Always use triple equals in javascript :) Double equals does weird coercion that you might not always expect